mad-dog/src/helpers.js

```diff
@@ -47,71 +38,130 @@ const USER_PIC_PATH = path.resolve('assets/images/profile-pic.jpg')
+  userCount,
+  executeAll,
+  executeOne
+) => {
+  const addedUserIds = []
+  const existingUserIds = []
+  const walletIndexToUserIdMap = {}
+
+  await _addUsers({ userCount, executeAll, executeOne, existingUserIds, addedUserIds, walletIndexToUserIdMap, uploadProfilePic: false })
+
+  // Map out walletId index => userId
+  return walletIndexToUserIdMap
+}
+
+/**
+ * Helper function to add a user. If the wallet index is already used to create a user, add that userId to
+ * the walletIndexToUserMap.
+ * @param {*} userCount
+ * @param {*} executeAll
+ * @param {*} executeOne
+ * @param {int[]} existingUserIds
+ * @param {int[]} addedUserIds
+ * @param {Object} walletIndexToUserIdMap
+ * @param {boolean} uploadProfilePic flag to upload profile pic on sign up or not
+ */
+async function _addUsers ({ userCount, executeAll, executeOne, existingUserIds, addedUserIds, walletIndexToUserIdMap, uploadProfilePic = true }) {
   await logOps('Add users', async () => {
     const users = genRandomUsers(userCount)
     await executeAll(async (libs, i) => {
+      try {
       // If user already exists, do not recreate user and use existing user
-      const existingUser = await getUser({ executeOne, walletIndex: i })
-      let userId
-      if (existingUser) {
-        existingUserIds.push(existingUser.user_id)
-        userId = existingUser.user_id
-        logger.info(`Found existing user: ${userId}`)
-      } else {
-        const userMetadata = users[i]
-        const newUserId = await addUser(libs, userMetadata, USER_PIC_PATH)
-        addedUserIds.push(newUserId)
-        userId = newUserId
-        logger.info(`Created new user: ${userId}`)
+        const existingUser = await getUser({ executeOne, walletIndex: i })
+        let userId
+        if (existingUser) {
+          logger.info(`Found existing user: ${existingUser.user_id}`)
+          existingUserIds.push(existingUser.user_id)
+          userId = existingUser.user_id
+        } else {
+          logger.info('Creating new user...')
+          const userMetadata = users[i]
+          let newUserId
+          if (uploadProfilePic) {
+            newUserId = await uploadProfileImagesAndAddUser(libs, userMetadata, USER_PIC_PATH)
+          } else {
+            newUserId = await addUser(libs, userMetadata)
+          }
+          logger.info(`Created new user: ${newUserId}`)
+          addedUserIds.push(newUserId)
+          userId = newUserId
+        }
 
         // Wait 1 indexing cycle to get all proper and expected user metadata, as the starter metadata
         // does not contain all necessary fields (blocknumber, track_blocknumber, ...)
         await waitForIndexing()
         const userWalletAddress = getLibsWalletAddress(libs)
         const userAccount = await getUserAccount(libs, userWalletAddress)
-        setCurrentUser(libs, userAccount)
-      }
+        setCurrentUserAndUpdateLibs(libs, userAccount) // might not need this anymore bc refactored signup flow
```

Comment:
yeah, comment seems correct

Reply:
TODO - lets confirm that this is the case and then remove, either way this code comment should be handled

Reply:
did a quick test and I actually do think this check is necessary. will revisit in refactoring mad dog but will leave in for now.